api/integrations/launch_darkly/services.py

```diff
@@ -164,31 +171,49 @@ def _create_mv_feature_states(
     environments_by_ld_environment_key: dict[str, Environment],
 ) -> None:
     variations = ld_flag["variations"]
+    variation_values_by_idx: dict[str, str] = {}
     mv_feature_options_by_variation: dict[str, MultivariateFeatureOption] = {}
 
     for idx, variation in enumerate(variations):
+        variation_idx = str(idx)
+        variation_value = variation["value"]
+        variation_values_by_idx[variation_idx] = variation_value
         (
             mv_feature_options_by_variation[str(idx)],
             _,
         ) = MultivariateFeatureOption.objects.update_or_create(
             feature=feature,
-            string_value=variation["value"],
+            string_value=variation_value,
             defaults={"default_percentage_allocation": 0, "type": STRING},
         )
 
     for ld_environment_key, environment in environments_by_ld_environment_key.items():
         ld_flag_config = ld_flag["environments"][ld_environment_key]
+
+        is_flag_on = ld_flag_config["on"]
+
         feature_state, _ = FeatureState.objects.update_or_create(
             feature=feature,
             environment=environment,
-            defaults={"enabled": ld_flag_config["on"]},
+            defaults={"enabled": is_flag_on},
         )
 
         cumulative_rollout = rollout_baseline = 0
 
         if ld_flag_config_summary := ld_flag_config.get("_summary"):
             enabled_variations = ld_flag_config_summary.get("variations") or {}
             for variation_idx, variation_config in enabled_variations.items():
+                if variation_config.get("isOff"):
+                    # Set LD's off value as the control value.
+                    # We expect only one off variation.
+                    FeatureStateValue.objects.update_or_create(
+                        feature_state=feature_state,
+                        defaults={
+                            "type": STRING,
+                            "string_value": variation_values_by_idx[variation_idx],
```

Comment:
So I guess here we're expecting that the variations are in the same order in both `ld_flag["variations]` and `ld_flag["environments"][ld_environment_key]["_summary"]["variations"]`. This does seem like a fairly safe assumption but might not be guaranteed, right? Is there some other key we can use to guarantee we're using the correct variation?

Reply:
I agree it's cringy, but it's the officially supported way. See the docs [documentation](https://apidocs.launchdarkly.com/tag/Feature-flags#section/Anatomy-of-a-feature-flag) (under the "Individual context targets" heading)

No idea why wouldn't they just use ids